### PR TITLE
Fix unicode character uri bug

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Fixes https://github.com/PowerShell/vscode-powershell/issues/2324

This removes  any usage of `Uri.ToString()` and creates a `worspaceService.GetFile(` that takes a `Uri` to better abstract this away.

This also revs the Omnisharp lib that contains my fix that also removes usages of `Uri.ToString()`

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4770933&bid=15899418)

File: src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs [hunk 1]
```diff
@@ -214,10 +208,7 @@ public bool TryGetFile(string filePath, out ScriptFile scriptFile)
         /// </summary>
         /// <param name="filePath">The file path for which a buffer will be retrieved.</param>
         /// <returns>A ScriptFile instance if there is a buffer for the path, null otherwise.</returns>
-        public ScriptFile GetFileBuffer(string filePath)
-        {
-            return this.GetFileBuffer(filePath, null);
-        }
+        public ScriptFile GetFileBuffer(string filePath) => GetFileBuffer(filePath, null);
```
Comment (rjmholt) [inline src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs]:
```suggestion
        public ScriptFile GetFileBuffer(string filePath) => GetFileBuffer(filePath, initialBuffer: null);
```

File: src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs [hunk 2]
```diff
@@ -226,27 +217,46 @@ public ScriptFile GetFileBuffer(string filePath)
...
         /// <returns>A ScriptFile instance for the specified path.</returns>
-        public ScriptFile GetFileBuffer(string filePath, string initialBuffer)
+        public ScriptFile GetFileBuffer(string filePath, string initialBuffer) => GetFileBuffer(new Uri(filePath), initialBuffer);
+
+        /// <summary>
+        /// Gets a new ScriptFile instance which is identified by the given file path.
+        /// </summary>
+        /// <param name="fileUri">The file Uri for which a buffer will be retrieved.</param>
+        /// <returns>A ScriptFile instance if there is a buffer for the path, null otherwise.</returns>
+        public ScriptFile GetFileBuffer(Uri fileUri) => GetFileBuffer(fileUri, null);
```
Comment (rjmholt) [inline src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs]:
```suggestion
        public ScriptFile GetFileBuffer(Uri fileUri) => GetFileBuffer(fileUri, initialBuffer: null);
```

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- src/PowerShellEditorServices/Services/Workspace/WorkspaceService.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4770933&bid=15899418)

**Pull request merged**